backend/ibutsu_server/widgets/run_aggregator.py

```diff
@@ -10,43 +14,45 @@ def _get_recent_run_data(weeks, group_field, project=None):
     delta = timedelta(weeks=weeks).total_seconds()
     current_time = time.time()
     time_period_in_sec = current_time - delta
-    # now do the aggregation based on the 'group_field'
-    pipeline = [
-        {
-            "$match": {
-                "start_time": {"$gte": time_period_in_sec},
-                f"{group_field}": {"$exists": "true", "$ne": "null"},
-            }
-        },
-        {
-            "$group": {
-                "_id": f"${group_field}",
-                "tests": {"$sum": "$summary.tests"},
-                "failures": {"$sum": "$summary.failures"},
-                "skips": {"$sum": "$summary.skips"},
-                "errors": {"$sum": "$summary.errors"},
-            }
-        },
-    ]
+
+    # create filters for start time and that the group_field exists
+    filters = [f"start_time>{time_period_in_sec}", f"{group_field}@y"]
     if project:
-        pipeline[0]["$match"]["metadata.project"] = {"$eq": project}
-
-    aggr = mongo.runs.aggregate(pipeline)
-
-    # some data cleanup
-    for i, result in enumerate(aggr):
-        # get rid of the `None`
-        if not result["_id"]:
-            continue
-        _id = result["_id"]
-        data["failed"][_id] = int(round(100 * float(result["failures"]) / float(result["tests"])))
-        data["skipped"][_id] = int(round(100 * float(result["skips"]) / float(result["tests"])))
-        data["error"][_id] = int(round(100 * float(result["errors"]) / float(result["tests"])))
-        data["passed"][_id] = int(
-            100 - (data["failed"][_id] + data["skipped"][_id] + data["error"][_id])
+        filters.append(f"metadata.project={project}")
+
+    # generate the group field
+    group_fields = group_field.split(".")
+    group_field = Run.data
+    for group in group_fields:
+        group_field = group_field[group]
+
+    # create the query
+    query = session.query(
+        group_field,
+        func.sum(Run.data["summary"]["failures"].cast(sqlalchemy.Float)),
```

Comment:
In `ibutsu_server.db.base` alias `Float` from the `db` object (like the others), and then import that class.

Reply:
Done!